src/ucp/core/ucp_request.c

```diff
@@ -264,6 +264,84 @@ UCS_PROFILE_FUNC_VOID(ucp_request_memory_dereg,
     }
 }
 
+ucs_status_t ucp_request_rndv_buffer_reg(ucp_request_t *req)
+{
+    ucp_ep_t        *ep = req->send.ep;
+    uct_md_h         md;
+    ucp_lane_index_t lane;
+    ucp_lane_index_t i;
+    ucp_lane_index_t r;
+    ucs_status_t     status;
+
+    req->send.reg_rsc = UCP_NULL_RESOURCE;
+
+    for (i = 0; i < ucp_ep_rndv_num_lanes(ep); i++) {
+        if (ucp_ep_rndv_md_flags(ep, i) & UCT_MD_FLAG_NEED_MEMH) {
+            lane = ucp_ep_get_rndv_get_lane(ep, i);
+            ucs_assert(lane != UCP_NULL_LANE);
+            md = ucp_ep_md(ep, lane);
+            status = uct_md_mem_reg(md, (void*)req->send.buffer, req->send.length,
+                                    UCT_MD_MEM_ACCESS_RMA,
+                                    &req->send.state.dt.dt.contig[i].memh);
+            if (status != UCS_OK) {
+                /* rollback all registrations */
+                for (r = 0; r < i; r++) {
+                    if (ucp_ep_rndv_md_flags(ep, r) & UCT_MD_FLAG_NEED_MEMH) {
+                        ucs_assert(req->send.state.dt.dt.contig[r].memh != UCT_MEM_HANDLE_NULL);
+                        lane = ucp_ep_get_rndv_get_lane(ep, r);
+                        ucs_assert(lane != UCP_NULL_LANE);
```

Comment:
no need for this assert

Reply:
assert helps to take in mind limitations, regular (not _always) asserts are removed in release build
@yosefe what do you think? remove it?

Reply:
BTW, let's ask Mike :)
@miked-mellanox  what do you think about asserts?

Reply:
agree w/ @hoopoepg - good way to remember limitations.

Reply:
the point here is that `ucp_ep_get_rndv_get_lane(ep, r)` already contains exactly the same assert. And then we add one more right after the call to `ucp_ep_get_rndv_get_lane(ep, r);`